File: bond_pricing/bond.py
import bond  

def price_zero(face_value: float, rate: float, years: float) -> float:
    """
    Price a zero-coupon bond.

    Parameters:
        face_value (float): The bond's face (par) value.
        rate (float): Annual interest rate (as a decimal).
        years (float): Time to maturity in years.

    Returns:
        float: Present value of the bond.
    """
    return bond.price_zero(face_value, rate, years)

def price_coupon(face_value: float, coupon: float, rate: float, periods: int) -> float:
    """
    Price a fixed-rate coupon bond.

    Parameters:
        face_value (float): The bond's face (par) value.
        coupon (float): The annual coupon payment.
        rate (float): Annual interest rate (as a decimal).
        periods (int): Number of periods until maturity.

    Returns:
        float: Present value of the bond.
    """
    return bond.price_coupon(face_value, coupon, rate, periods)

File: bond_pricing/bond.c
#include <math.h>

// Price of a zero-coupon bond: PV = FV / (1 + r)^t
double price_zero_coupon(double face_value, double rate, double years) {
    return face_value / pow(1.0 + rate, years);
}

// Price of a coupon bond: PV = Σ (C / (1 + r)^t) + FV / (1 + r)^T
double price_coupon_bond(double face_value, double coupon, double rate, int periods) {
    double pv = 0.0;
    for (int t = 1; t <= periods; t++) {
        pv += coupon / pow(1.0 + rate, t);
    }
    pv += face_value / pow(1.0 + rate, periods);
    return pv;
}